mm2src/common/mm_number.rs

```diff
@@ -86,16 +93,24 @@ impl Mul for &MmNumber {
     fn mul(self, rhs: Self) -> Self::Output {
         let lhs: BigRational = self.clone().into();
         let rhs: BigRational = rhs.clone().into();
-        MmNumber::from(lhs * rhs)
+        (lhs * rhs).into()
     }
 }
 
 impl Add for MmNumber {
     type Output = MmNumber;
 
     fn add(self, rhs: Self) -> Self::Output {
-        let lhs: BigRational = self.into();
-        let rhs: BigRational = rhs.into();
+        (self.0 + rhs.0).into()
+    }
+}
+
+impl Add for &MmNumber {
+    type Output = MmNumber;
+
+    fn add(self, rhs: Self) -> Self::Output {
+        let lhs: BigRational = self.clone().into();
```

Comment:
May we avoid clone here?

Reply:
And there're more places where it can be potentially avoided.

Reply:
Removed excess clone